tests/blast/test_blast_input.h

```diff
@@ -0,0 +1,1166 @@
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Hannes Hauswedell <hannes.hauswedell@fu-berlin.de>
+// ==========================================================================
+// Tests for the blast module
+// ==========================================================================
+
+#include <fstream>
+
+// Files that are being read by this implementation
+#define PLUS_HEADER_DEFAULTS   "/tests/blast/plus_header_defaults.blast"
+#define LEGACY_HEADER_DEFAULTS "/tests/blast/legacy_header_defaults.blast"
+#define PLUS_HEADER_CUSTOM     "/tests/blast/plus_header_custom.m9"
+// same for HEADER and NOHEADERS:
+#define NOHEADER_DEFAULTS      "/tests/blast/noheader_defaults.m8"
+// same for HEADER and NOHEADERS:
+#define NOHEADER_CUSTOM        "/tests/blast/noheader_custom.m8"
+
+using namespace seqan;
+
+template <typename TNum,
+          typename std::enable_if<Is<NumberConcept<TNum>>::VALUE, int>::type = 0>
+inline void
+clear(TNum & num)
+{
+    num = 0;
+}
+
+template <typename TArg,
+          typename... TArgs>
+inline void
+clear(TArg & arg, TArgs & ... args)
+{
+    clear(arg);
+    clear(args...);
+}
+
+void _test_blast_read_tabular_match(std::string const & path,
+                                    bool const defaults,
+                                    bool const withheader,
+                                    bool const islegacy)
+{
+    // only used when defaults == false
+    typedef BlastMatchField<> TField;
+
+    std::string inPath = std::string(SEQAN_PATH_TO_ROOT()) + std::string(path);
+
+    std::ifstream ifstream(toCString(inPath),
```

Comment:
Hm, I can change this, but then we loose a lot of currently tested functionality...